src/endpoints/create.rs

```diff
@@ -278,12 +282,20 @@ pub async fn create(
         }
     }
 
-    if ARGS.readonly && ARGS.uploader_password.is_some() {
-        if uploader_password != ARGS.uploader_password.as_ref().unwrap().to_owned() {
-            return Ok(HttpResponse::Found()
-                .append_header(("Location", format!("{}/incorrect", ARGS.public_path_as_str())))
-                .finish());
-        }
+    let res = axum::response::Response::builder()
+        .status(302)
+        .header(
+            "Location",
+            format!("{}/incorrect", ARGS.public_path_as_str()),
+        )
+        .body("".to_string())
+        .unwrap();
+
+    if ARGS.readonly
```

Comment:
have you tried the new let chains feature?
(see: https://blog.rust-lang.org/2025/06/26/Rust-1.88.0/#let-chains)

Should allow you to get rid of the `.unwrap` here